src/number_theory/cyclotomic/primitive_roots.lean

```diff
@@ -161,21 +162,34 @@ section norm
 namespace is_primitive_root
 
 variables [field L] {ζ : L} (hζ : is_primitive_root ζ n)
+variables {K} [field K] [algebra K L] [ne_zero ((n : ℕ) : K)]
+
+/-- If `finrank K L = 1`, then the norm of a `2`-th primitive root of unity is `-1`. This
+mathematically trivial result is complementary to `norm_eq_one` below. -/
+lemma norm_eq_neg_one [field K] [algebra K L] [is_cyclotomic_extension {2} K L]
+  (h : finrank K L = 1) (hζ : is_primitive_root ζ 2) : norm K ζ = -1 :=
+by rw [is_primitive_root.eq_neg_one_of_two_right hζ, show -1 = algebra_map K L (-1), by simp,
+      norm_algebra_map, h, pow_one]
 
 include hζ
 
-/-- If `K` is linearly ordered (in particular for `K = ℚ`), the norm of a primitive root is `1`
-if `n` is odd. -/
-lemma norm_eq_one [linear_ordered_field K] [algebra K L] (hodd : odd (n : ℕ)) : norm K ζ = 1 :=
+/-- If `irreducible (cyclotomic n K)` (in particular for `K = ℚ`), the norm of a primitive root is
+`1` if `n ≠ 2`. -/
+lemma norm_eq_one [field K] [ne_zero ((n : ℕ) : K)] [algebra K L] [is_cyclotomic_extension {n} K L]
```

Comment:
I liked Kevin's idea that we just make the result `(-1) ^ p-1`; is this too difficult to do? Maybe also provide a lemma that this is basically `if p = 2 then -1 else 1` for a prime

Reply:
`n` is any `pnat`, not necessarily prime, so `(-1) ^ (n - 1)` is not the right result. It's really `-1` if `n=2` and `1` otherwise. Maybe a result with a `ite`?

Reply:
I've added `norm_of_cyclotomic_irreducible`.